app/javascript/mastodon/components/hover_card_controller.tsx

```diff
@@ -48,53 +67,74 @@ export const HoverCardController: React.FC = () => {
       if (target === cardRef.current?.parentNode) {
         cancelLeaveTimeout();
       }
-    },
-    [cancelLeaveTimeout, setEnterTimeout, setOpen, setAccountId, setAnchor],
-  );
+    };
 
-  const handleAnchorMouseLeave = useCallback(
-    (e: MouseEvent) => {
-      if (e.target === anchor || e.target === cardRef.current?.parentNode) {
+    const handleMouseLeave = (e: MouseEvent) => {
+      if (
+        e.target === anchorRef.current ||
+        e.target === cardRef.current?.parentNode
+      ) {
         cancelEnterTimeout();
 
         setLeaveTimeout(() => {
-          anchor?.removeAttribute('aria-describedby');
+          anchorRef.current?.removeAttribute('aria-describedby');
           setOpen(false);
           setAnchor(null);
         }, leaveDelay);
       }
-    },
-    [cancelEnterTimeout, setLeaveTimeout, setOpen, setAnchor, anchor],
-  );
+    };
 
-  const handleClose = useCallback(() => {
-    cancelEnterTimeout();
-    cancelLeaveTimeout();
-    setOpen(false);
-    setAnchor(null);
-  }, [cancelEnterTimeout, cancelLeaveTimeout, setOpen, setAnchor]);
+    const handleScrollEnd = () => {
+      scrollingRef.current = false;
+    };
 
-  useEffect(() => {
-    handleClose();
-  }, [handleClose, location]);
+    const handleScroll = () => {
+      scrollingRef.current = true;
+      cancelEnterTimeout();
+      setScrollTimeout(handleScrollEnd, 100);
```

Comment:
I am not sure if I understand it correctly, but you never cancel the previous scroll timeout, so it will trigger?

I have a feeling you want a debounce here, to call `scrollingRef.current = false` 100ms after the last scroll? If yes, maybe you could use the https://github.com/xnimorz/use-debounce package that I introduced in the grouped notifications PR?

But it would not be compatible with my idea above of not having it in a ref.

Reply:
`setScrollTimeout` cancels the previous timeout if there is one. Basically, 100ms after the last scroll event `scrollingRef.current` will be set to `false`.